datalad/customremotes/tests/test_archives.py

```diff
@@ -338,3 +338,48 @@ def raise_AttributeError(*args):
         assert_equal(f.read(), "LOAD")
     assert_equal(stats(tempfile, times=False), stats(tempfile2, times=False))
     unlink(tempfile2)  # TODO: next two with_tempfile
+
+
+@with_tree(tree=
+    {'1.tar.gz':
+         {
+             'bu.dat': '52055957098986598349795121365535' * 10000,
+             'bu3.dat': '8236397048205454767887168342849275422' * 10000
+          },
+    }
+)
+@with_tempfile
+def test_autoclean_cache(dspath, clonepath):
+    from datalad.consts import ARCHIVES_TEMP_DIR
+    from datalad.api import add_archive_content
+    from datalad.api import create
+    from datalad.api import clone
+
+    ds = create(dspath, force=True)
+    ds.save()
+    # NOTE: add_archive_content isn't proper datalad command ATM. It doesn't
+    # operate on a Dataset instance!
+    with chpwd(dspath):
+        add_archive_content('1.tar.gz', delete=True, drop_after=True)
+
+    clone_ds = clone(ds.path, clonepath)
+    cache_path = clone_ds.pathobj / ARCHIVES_TEMP_DIR
+
+    try:
+        with patch.dict("os.environ",
+                        {"DATALAD_REMOTE_ARCHIVES_AUTOCLEAN": "1"}):
+            dl_cfg.reload()
+            clone_ds.get()
```

Comment:
Since we haven't tested ever functionality with such non-persistent cache, I would recommend
- make archive contain lots (100?) of (small) files
- add `jobs=5` or so here to see if there is no some kind of interaction between separate get jobs while trying to use non-persistent cache

Reply:
Ok, thanks!